labgrid/pytestplugin/hooks.py

```diff
@@ -1,30 +1,66 @@
-import logging
 import os
 import warnings
+import logging
 import pytest
 
 from .. import Environment
 from ..consoleloggingreporter import ConsoleLoggingReporter
-from .reporter import StepReporter, ColoredStepReporter
 from ..util.helper import processwrapper
+from ..logging import StepFormatter, StepLogger
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_cmdline_main(config):
+    def set_cli_log_level(l):
+        nonlocal config
+        current_level = getattr(config.option, "log_cli_level", None)
+
+        if isinstance(current_level, str):
+            current_level = current_level.upper()
+            try:
+                current_level = int(getattr(logging, current_level, None))
+            except ValueError:
+                current_level = logging.WARNING
+
+        if current_level is None or \
+                current_level == logging.NOTSET or \
+                current_level > l:
+            config.option.log_cli_level = str(l)
```

Comment:
```suggestion
            try:
                current_level = int(logging.getLevelName(current_level))
            except ValueError:
                current_level = logging.NOTSET

        if current_level == logging.NOTSET or current_level > l:
            config.option.log_cli_level = str(l)
```
As far as I understand the [docs](https://docs.pytest.org/en/7.1.x/reference/reference.html#confval-log_cli_level), we don't need to (and shouldn't) handle lower case log level names.